assignments/this.js

```diff
@@ -13,14 +13,60 @@
 
 // code example for Window Binding
 
+window._ = {};
+
 // Principle 2
 
 // code example for Implicit Binding
 
+function foo() {
+    console.log(this.a);
+}
+
+const obj2 = {
+    a: 10,
+    foo: foo
+};
+
+const obj1 = {
+    a: 4,
+    obj2: obj2
+};
+
+obj1.obj2.foo();
+
 // Principle 3
 
 // code example for New Binding
 
+function foo(a) {
+    this.a = a;
+}
+
+var bar = new foo ( 2 );
+console.log( bar.a );
+
 // Principle 4
 
-// code example for Explicit Binding
\ No newline at end of file
+// code example for Explicit Binding
+
+function foo(something) {
+    console.log( this.a, something );
```

Comment:
this will log 'undefined, 2, 2 3
please go back and take a look at this one to see how it can be fixed to print out the proper values.